lib/Onyx.js

```diff
@@ -181,6 +181,64 @@ function isSafeEvictionKey(testKey) {
     return _.some(evictionAllowList, key => isKeyMatch(key, testKey));
 }
 
+/**
+ * Tries to return all keys
+ * from the current cache.
+ * (internal)
+ * @returns {Array<string>}
+ */
+function getAllKeysSync() {
+    return cache.getAllKeys();
+}
+
+/**
+ * Tries to get a value from the cache.
+ * If the value is not present in cache it will
+ * return the default value or undefined.
+ * If the requested key is a collection, it will
+ * return an object with all the collection members.
+ *
+ * @param {string} key
+ * @param {Object} mapping
+ * @returns {string}
+ */
+function tryGetCachedValue(key, mapping = {}) {
+    if (isCollectionKey(key)) {
+        const allKeys = getAllKeysSync();
+        const matchingKeys = _.filter(allKeys, k => k.startsWith(key));
+        const values = _.reduce(matchingKeys, (finalObject, matchedKey) => {
+            const val = cache.getValue(matchedKey);
+            if (val) {
+                // We know a 100% where the object is coming from, so its fine
+                // eslint-disable-next-line no-param-reassign
+                finalObject[matchedKey] = val;
+            }
+            return finalObject;
+        }, {});
+        if (_.keys(values).length > 0) {
+            return values;
+        }
+    }
+
+    if (!cache.hasCacheForKey(key)) {
+        return defaultKeyStates[key]; // We might have the value in the default key state
+    }
+
+    const val = cache.getValue(key);
+
+    // If the mapping has a selector, then the component's state must only be updated with the data
+    // returned by the selector.
```

Comment:
What were we doing previously? Returning additional data from outside the selector?

Reply:
I removed the comment as it seems to be ambiguous. As this logic is new there is no previous behaviour, and the comment was just trying to say that we need to use the selector which is redundant